src/modules/RA/BaseLayer/components/BaseLayerForm.js

```diff
@@ -0,0 +1,172 @@
+  const logMin = Math.log2(min);
+  const logMax = Math.log2(max);
+  return Array.from({ length: logMax - logMin + 1 }, ((_, index) => {
+    const number = 2 ** (logMin + index);
+    return {
+      id: number,
+      name: number.toString(),
+    };
+  }));
+};
+
+
+const BaseLayerForm = ({ basePath, classes, translate: t, ...props }) => {
+  const { record: { base_layer_type: layerType } } = props;
+
+  const [currentLayerType, setLayerType] = useState(layerType);
+
+  const tileSizeInitialValue = currentLayerType === 'raster' ? 256 : 512;
+
+  const handleTypeChange = useCallback(({ target: { value } }) => {
+    setLayerType(value);
+  }, []);
+
+  const validateURL = React.useCallback(value => {
+    if (!value) {
+      return undefined;
+    }
+    return !value.match(/(^(?:mapbox:|https?:)?\/\/.)/)
+      ? t('baseLayer.form.errors.url')
+      : undefined;
+  }, [t]);
+
+  const validateTiles = React.useCallback(value => {
+    if (!value) {
+      return undefined;
+    }
+    const URLisValid = validateURL(value);
+    if (URLisValid) {
+      return URLisValid;
+    }
+    return (['{x}', '{y}', '{z}'].some(item => !value.includes(item)) && !value.includes('{bbox-epsg-3857}'))
+      ? t('baseLayer.form.errors.mandatoryTilesWord')
+      : undefined;
+  }, [t, validateURL]);
+
+  const validateGlyphs = React.useCallback(value => {
+    if (!value) {
+      return undefined;
+    }
+    const URLisValid = validateURL(value);
+    if (URLisValid) {
+      return URLisValid;
+    }
+    if (!value.includes('{fontstack}')) {
+      return t('baseLayer.form.errors.mandatoryWord', { name: '{fontstack}' });
+    }
+    if (!value.includes('{range}')) {
+      return t('baseLayer.form.errors.mandatoryWord', { name: '{range}' });
+    }
+    return undefined;
+  }, [t, validateURL]);
+
+  return (
+    <SimpleForm {...props}>
+      <TextInput source="name" label="baseLayer.form.name" validate={required()} />
+
+      <SelectInput
+        choices={[
+          { id: 'raster', name: t('baseLayer.form.type.choices.raster') },
+          { id: 'vector', name: t('baseLayer.form.type.choices.vector') },
+          { id: 'mapbox', name: t('baseLayer.form.type.choices.mapbox') },
+        ]}
+        disabled={Boolean(layerType)}
+        label="baseLayer.form.type.select"
+        onChange={handleTypeChange}
+        source="base_layer_type"
+        validate={required()}
+      />
+
+      {currentLayerType && (
+        <FieldGroup>
+          <SelectInput
+            choices={getTileSizeChoices(({ min: tileSizeInitialValue }))}
+            initialValue={tileSizeInitialValue}
+            label="baseLayer.form.tileSize"
+            source="tile_size"
+            translateChoice={false}
+            validate={required()}
+          />
+
+          {currentLayerType === 'mapbox' && (
+            <TextInput
+              label="baseLayer.form.mapBoxUrl"
+              source="map_box_url"
```

Comment:
```suggestion
              source="mapbox_url"
```

Reply:
It's anesthetic but I cannot change the value of the source